<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1174</width>
    <height>972</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>3DViewer</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="WidgetOGL" name="openGLWidget">
      <property name="minimumSize">
       <size>
        <width>400</width>
        <height>400</height>
       </size>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="controlPanel">
      <property name="maximumSize">
       <size>
        <width>384</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="title">
       <string/>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_8">
       <item>
        <widget class="QGroupBox" name="saving">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="title">
          <string>Открытие и сохранение</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_6">
          <item>
           <widget class="QPushButton" name="pushButton_selectFile">
            <property name="text">
             <string>Открыть файл</string>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_11">
            <item>
             <widget class="QPushButton" name="savePicture">
              <property name="enabled">
               <bool>true</bool>
              </property>
              <property name="text">
               <string>Сохранить картинку</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="saveGif">
              <property name="enabled">
               <bool>true</bool>
              </property>
              <property name="text">
               <string>Записать гифку</string>
              </property>
              <property name="checkable">
               <bool>true</bool>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="scaling">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="title">
          <string>Масштабирование</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_5">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_10">
            <item>
             <widget class="QPushButton" name="scaleToCenter">
              <property name="enabled">
               <bool>true</bool>
              </property>
              <property name="text">
               <string>Центрировать</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="scaleOptimize">
              <property name="enabled">
               <bool>true</bool>
              </property>
              <property name="text">
               <string>Вписать в экран</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_9">
            <item>
             <widget class="QPushButton" name="scaleDown">
              <property name="text">
               <string>&lt;&lt;</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSpinBox" name="scaleStep">
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>50</number>
              </property>
              <property name="value">
               <number>25</number>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="scaleUp">
              <property name="text">
               <string>&gt;&gt;</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="moving">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="title">
          <string>Перемещение</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_5">
            <item>
             <widget class="QRadioButton" name="moveX">
              <property name="text">
               <string>X</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="moveY">
              <property name="text">
               <string>Y</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="moveZ">
              <property name="text">
               <string>Z</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_4">
            <item>
             <widget class="QPushButton" name="moveBack">
              <property name="text">
               <string>&lt;&lt;</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSpinBox" name="moveStep">
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>100</number>
              </property>
              <property name="value">
               <number>15</number>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="moveForward">
              <property name="text">
               <string>&gt;&gt;</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="rotating">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="title">
          <string>Вращение</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_7">
            <item>
             <widget class="QRadioButton" name="rotateX">
              <property name="text">
               <string>X</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="rotateY">
              <property name="text">
               <string>Y</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="rotateZ">
              <property name="text">
               <string>Z</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_8">
            <item>
             <widget class="QPushButton" name="rotateBack">
              <property name="text">
               <string>&lt;&lt;</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSpinBox" name="rotateStep">
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>359</number>
              </property>
              <property name="value">
               <number>15</number>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="rotateForward">
              <property name="text">
               <string>&gt;&gt;</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="viewingArea">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="title">
          <string>Настройка области просмотра</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_7">
          <item>
           <widget class="QCheckBox" name="perspective">
            <property name="text">
             <string>Перспективная проекция</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="bgColorBtn">
            <property name="text">
             <string>Выбрать цвет фона</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="edgeSet">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="title">
          <string>Настройка рёбер</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout">
          <item>
           <widget class="QCheckBox" name="edgeDashChk">
            <property name="text">
             <string>Пунктир</string>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <item>
             <widget class="QLabel" name="edgeThickLbl">
              <property name="text">
               <string>Толщина</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSpinBox" name="edgeThickBox">
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>30</number>
              </property>
              <property name="value">
               <number>4</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QPushButton" name="edgeColorBtn">
            <property name="text">
             <string>Выбрать цвет</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="vertexSet">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="title">
          <string>Настройка вершин</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_6">
            <item>
             <widget class="QRadioButton" name="displayMethodNot">
              <property name="font">
               <font>
                <pointsize>11</pointsize>
               </font>
              </property>
              <property name="text">
               <string>Нет</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="displayMethodSphere">
              <property name="font">
               <font>
                <pointsize>11</pointsize>
               </font>
              </property>
              <property name="text">
               <string>Круг</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="displayMethodCube">
              <property name="font">
               <font>
                <pointsize>11</pointsize>
               </font>
              </property>
              <property name="text">
               <string>Квадрат</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_3">
            <item>
             <widget class="QLabel" name="vSizeLbl">
              <property name="text">
               <string>Размер</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSpinBox" name="vSizeBox">
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>60</number>
              </property>
              <property name="value">
               <number>25</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QPushButton" name="vColorBtn">
            <property name="text">
             <string>Выбрать цвет</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="emptyArea">
         <property name="title">
          <string/>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1174</width>
     <height>23</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>WidgetOGL</class>
   <extends>QOpenGLWidget</extends>
   <header>widgetogl.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
